src/elements/content-sidebar/withSidebarAnnotations.js

```diff
@@ -150,6 +199,43 @@ export default function withSidebarAnnotations(
             this.refreshActivitySidebar();
         }
 
+        deleteAnnotationReply() {
+            const {
+                annotatorState: {
+                    action,
+                    annotation: { id: annotationId },
+                    annotationReply: { id: replyId },
+                },
+                api,
+                file,
+            } = this.props;
+
+            const feedAPI = api.getFeedAPI(false);
+            feedAPI.file = file;
+
+            if (action === 'reply_delete_start') {
+                feedAPI.updateReplyItem({ isPending: true }, annotationId, replyId);
+            } else {
+                const { items: feedItems = [] } = feedAPI.getCachedItems(file.id) || {};
+                const annotationItem = feedItems.find(({ id }) => id === annotationId);
+
+                if (!annotationItem) {
+                    return;
+                }
+
+                // Check if the parent annotation has the reply currently visible
+                const replyItem = annotationItem.replies.find(({ id }) => id === replyId);
+                if (replyItem) {
+                    feedAPI.deleteReplyItem(replyId, annotationId);
+                } else if (annotationItem.total_reply_count !== undefined) {
```

Comment:
could total_reply_count be 0? If so, the logic below would make a negative I believe

Reply:
Good point, I've updated this condition.